### ISSUE TITLE
Problem With Forge and minecraft 1.10

### ISSUE THREAD

**Issue by Alacard** [itzg/dockerfiles#91]
JVM_OPTS=-Xmx5024M -Xms5024M
VERSION=1.10
FORGEVERSION=12.18.0.2000
TYPE=FORGE
EULA=TRUE

Above is what i have set as my variables but for some reason it dont load forge but just loads vanilla. tryed checking the log (via docker logs -f mc) but the log dont show any errors about loading forge only thing it shows is this

> Downloading minecraft_server.1.10.jar
> [20:24:23] [Server thread/INFO]: Starting minecraft server version 1.10
> [20:24:23] [Server thread/INFO]: Loading properties
> [20:24:23] [Server thread/INFO]: Default game type: SURVIVAL
> [20:24:23] [Server thread/INFO]: Generating keypair
> [20:24:23] [Server thread/INFO]: Starting Minecraft server on *:25565
> [20:24:23] [Server thread/INFO]: Using epoll channel type
> [20:24:23] [Server thread/WARN]: Failed to load user banlist: 
> java.io.FileNotFoundException: banned-players.json (No such file or directory)
>     at java.io.FileInputStream.open(Native Method) ~[?:1.7.0_65]
>     at java.io.FileInputStream.<init>(FileInputStream.java:146) ~[?:1.7.0_65]
>     at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.10.jar:?]
>     at mt.g(SourceFile:136) ~[minecraft_server.1.10.jar:?]
>     at lc.z(SourceFile:99) [minecraft_server.1.10.jar:?]
>     at lc.<init>(SourceFile:25) [minecraft_server.1.10.jar:?]
>     at ld.j(SourceFile:185) [minecraft_server.1.10.jar:?]
>     at net.minecraft.server.MinecraftServer.run(SourceFile:428) [minecraft_server.1.10.jar:?]
>     at java.lang.Thread.run(Thread.java:745) [?:1.7.0_65]
> [20:24:23] [Server thread/WARN]: Failed to load ip banlist: 
> java.io.FileNotFoundException: banned-ips.json (No such file or directory)
>     at java.io.FileInputStream.open(Native Method) ~[?:1.7.0_65]
>     at java.io.FileInputStream.<init>(FileInputStream.java:146) ~[?:1.7.0_65]
>     at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.10.jar:?]
>     at mt.g(SourceFile:136) ~[minecraft_server.1.10.jar:?]
>     at lc.y(SourceFile:91) [minecraft_server.1.10.jar:?]
>     at lc.<init>(SourceFile:27) [minecraft_server.1.10.jar:?]
>     at ld.j(SourceFile:185) [minecraft_server.1.10.jar:?]
>     at net.minecraft.server.MinecraftServer.run(SourceFile:428) [minecraft_server.1.10.jar:?]
>     at java.lang.Thread.run(Thread.java:745) [?:1.7.0_65]
> [20:24:23] [Server thread/WARN]: Failed to load operators list: 
> java.io.FileNotFoundException: ops.json (No such file or directory)
>     at java.io.FileInputStream.open(Native Method) ~[?:1.7.0_65]
>     at java.io.FileInputStream.<init>(FileInputStream.java:146) ~[?:1.7.0_65]
>     at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.10.jar:?]
>     at mt.g(SourceFile:136) ~[minecraft_server.1.10.jar:?]
>     at lc.A(SourceFile:107) [minecraft_server.1.10.jar:?]
>     at lc.<init>(SourceFile:29) [minecraft_server.1.10.jar:?]
>     at ld.j(SourceFile:185) [minecraft_server.1.10.jar:?]
>     at net.minecraft.server.MinecraftServer.run(SourceFile:428) [minecraft_server.1.10.jar:?]
>     at java.lang.Thread.run(Thread.java:745) [?:1.7.0_65]
> [20:24:23] [Server thread/WARN]: Failed to load white-list: 
> java.io.FileNotFoundException: whitelist.json (No such file or directory)
>     at java.io.FileInputStream.open(Native Method) ~[?:1.7.0_65]
>     at java.io.FileInputStream.<init>(FileInputStream.java:146) ~[?:1.7.0_65]
>     at com.google.common.io.Files.newReader(Files.java:86) ~[minecraft_server.1.10.jar:?]
>     at mt.g(SourceFile:136) ~[minecraft_server.1.10.jar:?]
>     at lc.C(SourceFile:123) [minecraft_server.1.10.jar:?]
>     at lc.<init>(SourceFile:30) [minecraft_server.1.10.jar:?]
>     at ld.j(SourceFile:185) [minecraft_server.1.10.jar:?]
>     at net.minecraft.server.MinecraftServer.run(SourceFile:428) [minecraft_server.1.10.jar:?]
>     at java.lang.Thread.run(Thread.java:745) [?:1.7.0_65]
> [20:24:23] [Server thread/INFO]: Preparing level "world"
> [20:24:23] [Server thread/INFO]: Preparing start region for level 0
> [20:24:24] [Server thread/INFO]: Preparing spawn area: 10%
> [20:24:25] [Server thread/INFO]: Preparing spawn area: 26%
> [20:24:26] [Server thread/INFO]: Preparing spawn area: 44%
> [20:24:27] [Server thread/INFO]: Preparing spawn area: 63%
> [20:24:28] [Server thread/INFO]: Preparing spawn area: 82%
> [20:24:29] [Server thread/INFO]: Done (6.276s)! For help, type "help" or "?"

**Comment by itzg** [comment]
I think you actually found a legit issue; however, my observed output is different. With the compose file:

```
version: '2'

services:
  minecraft:
    image: itzg/minecraft-server
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      JVM_OPTS: "-Xmx5024M -Xms5024M"
      VERSION: "1.10"
      TYPE: "FORGE"
```

It's not properly downloading (or something) the Forge file according to the output:

```
Recreating issue91_minecraft_1
Attaching to issue91_minecraft_1
minecraft_1  | usermod: no changes
minecraft_1  | Switching to user 'minecraft'
minecraft_1  | Checking version information.
minecraft_1  | Checking type information.
minecraft_1  | Checking Forge version information.
minecraft_1  | Downloading forge-1.10-12.18.0.2000-installer.jar ...
minecraft_1  | Installing forge-1.10-12.18.0.2000-universal.jar
minecraft_1  | Error: Unable to access jarfile forge-1.10-12.18.0.2000-installer.jar
minecraft_1  | Error: Unable to access jarfile forge-1.10-12.18.0.2000-universal.jar
```

**Comment by Snorremd** [comment]
Ran into the same problem as you itzg. Same output:

```
minecraft |
minecraft | Downloading forge-1.10--installer.jar ...
minecraft | Installing forge-1.10--universal.jar
minecraft | Error: Unable to access jarfile forge-1.10--installer.jar
minecraft | Error: Unable to access jarfile forge-1.10--universal.jar
```

**Comment by c650** [comment]
I can confirm this is still an issue.

**Comment by itzg** [comment]
@c650 , @Snorremd , @Alacard  it has taken me forever, but I finally have conquered the different URL and launcher variants especially to fix the download of 1.10. Please pull the latest `itzg/minecraft-server` and this should now work properly.
